### PR TITLE
Closes #59

### LINKED ISSUE #59: Add possibility to run tests on a new-namespace
Part of #55, depends on #57

Let's introduce a `yaks-config.yaml` file that, when present in a directory that is going to be run, can specify some options:

```
config:
  namespace:
    temporary: false
    autoRemove: true
```

The "temporary" option is set to `false` by default, "autoRemove" is true.
When the "temporary" option is set to `true`, before running the tests in the test group:
- A new namespace with a random name (starting with `yaks-`) is created
- Yaks is installed in that namespace with default options

After running all tests in the test group:
- If "autoRemove" is enabled (default), the temporary namespace is deleted
- Let's make sure that autoRemove only removes temporary namespaces if created

### PR DESCRIPTION
**Pull request by bouskaJ**

### CODE REVIEW

File: pkg/cmd/test.go [hunk 1]
```diff
@@ -75,63 +85,89 @@ type testCmdOptions struct {
...
+	var conf *config.TestConfig
+
 	results := make(map[string]error)
+	defer printSummary(results)
 
-	execute := func(name string) {
-		_, err := o.createTest(c, name)
-		results[name] = err
+	if conf, err = config.LoadConfig(configFile); err != nil {
+		panic(err)
```
Comment (nicolaferraro) [inline pkg/cmd/test.go]: Returning the error is better in this context.

File: pkg/cmd/test.go [hunk 2]
```diff
@@ -75,63 +85,89 @@ type testCmdOptions struct {
...
 
-	for _, source := range args {
-		if isRemoteFile(source) {
-			execute(source)
-			continue
+	if conf.Config.Namespace.Temporary {
+		namespaceName := "yaks-" + uuid.New().String()
+		var namespace metav1.Object
+		if namespace, err = initializeTempNamespace(namespaceName, c, o.Context); err != nil {
+			panic(err)
```
Comment (nicolaferraro) [inline pkg/cmd/test.go]: Also here

File: pkg/cmd/test.go [hunk 3]
```diff
@@ -75,63 +85,89 @@ type testCmdOptions struct {
...
+		}
+		if conf.Config.Namespace.AutoRemove {
+			defer deleteTempNamespace(namespace, c, o.Context)
+		}
+
+		o.Namespace = namespaceName
+
+		copy := o.RootCmdOptions
+		if err = newCmdInstall(copy).Execute(); err != nil {
+			panic(err)
```
Comment (nicolaferraro) [inline pkg/cmd/test.go]: return

File: pkg/cmd/test.go [hunk 4]
```diff
@@ -75,63 +85,89 @@ type testCmdOptions struct {
...
+	if len(args) != 1 {
+		return errors.New(fmt.Sprintf("accepts exactly 1 test name to execute, received %d", len(args)))
+	}
+
+	return nil
+}
+
 func (o *testCmdOptions) run(_ *cobra.Command, args []string) error {
+	var err error
+	source := args[0]
```
Comment (nicolaferraro) [inline pkg/cmd/test.go]: I think this no longer works with multiple test files, and panics with no test files passed.

Comment (nicolaferraro) [reply pkg/cmd/test.go]: We should add some tests to avoid breaking things

**Comment by bouskaJ** [comment]
> I cannot try it but I see something that can breaks other features (like running multiple files at once).

I had introduced running multiple files in https://github.com/citrusframework/yaks/commit/b6ff2e81594e51ff69802e37c5d7c37512ccf4c7#diff-93aaaa55867aff8fd9da6d394ee1a81aL75-L82 I decided to remove this feature because it could be confusing. One file could run in `namespace1` and the other in `namespace2`. Isn't it misleading during searching for test results?
> 
> We should add some tests and personally I'd organize the code in two main functions: `runTest` and `runGroup` because they have different logic that when put together in a single location can create confusion.

> 
> The main run function can scan the arguments and call `runTest` or `runGroup` depending on the argument passed, then collect the results and print the summary.
> 
> `runGroup` can recursively call itself (later) or call `runTest`.
> 
> Wdyt?
> 
> Adding tests and doc is a lot important here.

Agree. I will split functions as well as add tests tomorrow.

**Comment by nicolaferraro** [comment]
> > I cannot try it but I see something that can breaks other features (like running multiple files at once).
> 
> I had introduced running multiple files in [b6ff2e8#diff-93aaaa55867aff8fd9da6d394ee1a81aL75-L82](https://github.com/citrusframework/yaks/commit/b6ff2e81594e51ff69802e37c5d7c37512ccf4c7#diff-93aaaa55867aff8fd9da6d394ee1a81aL75-L82) I decided to remove this feature because it could be confusing. One file could run in `namespace1` and the other in `namespace2`. Isn't it misleading during searching for test results?
> 
> > We should add some tests and personally I'd organize the code in two main functions: `runTest` and `runGroup` because they have different logic that when put together in a single location can create confusion.
> 
> > The main run function can scan the arguments and call `runTest` or `runGroup` depending on the argument passed, then collect the results and print the summary.
> > `runGroup` can recursively call itself (later) or call `runTest`.
> > Wdyt?
> > Adding tests and doc is a lot important here.
> 
> Agree. I will split functions as well as add tests tomorrow.

Yeah, sounds good removing the possibility of running multiple files now that we have groups. It's better having some validation that prints a user friendly message when one specifies 0 or 2+ files.

**Comment by christophd** [comment]
@bouskaJ can you please have a look what is missing here for this PR to be merged? thx!

**Comment by bouskaJ** [comment]
@christophd I am definitely going to incorporate changes but I am afraid I will not have time for this in the next two weeks.

**Comment by bouskaJ** [comment]
I have included Nicola's feedback. The remaining part is testing.

**Pull request merged**